crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -3158,6 +3159,14 @@ impl AuthorityPerEpochStore {
             }
         }
 
+        {
+            let mut deferred_transactions =
+                self.consensus_output_cache.deferred_transactions.lock();
+            for deleted_deferred_key in output.get_deleted_deferred_txn_keys() {
+                deferred_transactions.remove(&deleted_deferred_key);
+            }
+        }
+
         self.consensus_quarantine
             .write()
             .push_consensus_output(output, self)?;
```

Comment:
Is it feasible/worth moving all output cache updates inside this function so as to prevent any future similar issues?

Reply:
the other two items in `ConsensusOutputCache` are removed outside of consensus handler at the point where they are read. shared_version_assignments is removed after checkpoint finalization and user_signatures_for_checkpoints is removed by checkpoint builder.